src/store/options.js

```diff
@@ -75,14 +71,44 @@ const Options = {
 
 export default Options;
 
-chrome.runtime.onMessage.addListener((message) => {
-  if (isUpdateOptionsMessage(message)) {
+const observers = new Set();
+export async function observe(property, fn) {
+  let value;
+  const wrapperFn = (options) => {
+    if (value === undefined || options[property] !== value) {
+      value = options[property];
+      fn(value);
+    }
+  };
+
+  observers.add(wrapperFn);
+
+  // Run for the first time with the current options
+  const options = await store.resolve(store.get(Options));
+  wrapperFn(options);
```

Comment:
If the first call throws, it is no longer possible to remove the observer (from the "overservers" set). 

When I think about it, the contract should be that "fn" must never throw. If it still does, perhaps deleting it from the observer list immediately and then rethrowing the exception makes sense here?

Maybe it makes to

Reply:
If getting options fails, than we can turn off the extension :P

Reply:
I meant calling wrapperFn, which calls "fn" can fail. But OK.

Reply:
Ok, i can move the try/catch from line 106 to line 80, so we will have one try/catch and both cases covered. Everybody happy, what do you think?